PostHog/posthog
# fix: clean up two factor

## Changes

There's currently a race condition introduced here https://github.com/PostHog/posthog/pull/26813/files that may double trigger the set up call and make it so the QR code being rendered doesn't match the token stored in the session. 

This PR also cleans up the set up code so the modal is only rendered once and not in 3 locations. 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

It doesn't have an impact.

## How did you test this code?

Manually.

I'd love for the review to pull this locally and test all 3 cases to make sure it's all working
1. Set up 2FA from profile
2. Set up 2FA from members list (click on 2FA not enabled tag)
3. Turn on the 2FA enforcement on the organization, reload and set it up via the forced modal.
